packages/neon/neon/lib/src/blocs/unified_search.dart

```diff
@@ -0,0 +1,160 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:meta/meta.dart';
+import 'package:neon/models.dart';
+import 'package:neon/src/bloc/bloc.dart';
+import 'package:neon/src/bloc/result.dart';
+import 'package:neon/src/blocs/apps.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:rxdart/rxdart.dart';
+
+abstract interface class UnifiedSearchBlocEvents {
+  void search(final String term);
+
+  void enable();
+
+  void disable();
+}
+
+abstract interface class UnifiedSearchBlocStates {
+  BehaviorSubject<bool> get enabled;
+
+  BehaviorSubject<Result<Map<CoreUnifiedSearchProvider, Result<CoreUnifiedSearchResult>>?>> get results;
+}
+
+@internal
+class UnifiedSearchBloc extends InteractiveBloc implements UnifiedSearchBlocEvents, UnifiedSearchBlocStates {
+  UnifiedSearchBloc(
+    this._appsBloc,
+    this._account,
+  ) {
+    _appsBloc.activeApp.listen((final _) {
+      if (enabled.value) {
+        disable();
+      }
+    });
+  }
+
+  final AppsBloc _appsBloc;
+  final Account _account;
+  String _term = '';
```

Comment:
I think nullable makes sense here

Reply:
I used empty string because that is also how the search is reset. If there was null but also empty string then there would be two "empty" possibilities.